hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/protocol/commands/CreatePipelineCommand.java

```diff
@@ -48,16 +52,37 @@ public CreatePipelineCommand(final PipelineID pipelineID,
     this.factor = factor;
     this.type = type;
     this.nodelist = datanodeList;
+    if (datanodeList.size() ==
+        XceiverServerRatis.DEFAULT_PRIORITY_LIST.size()) {
```

Comment:
just forward to the Ctor with `priorityList `.

```
this(pipelineID, factor, type, datanodeList,
    new ArrayList<>(Collections.nCopies(datanodeList.size(), 0))
```

Reply:
always new is a waste.